### ISSUE TITLE
How to pass Json using command line parser ?

### ISSUE THREAD

**Issue by sybaris** [commandlineparser/commandline#876]
Hi,

I would like to create a console program like 
`Invoke-RestMethod -Body $body`
to pass a **json content** using **powershell** on windows.

```
[string] $body = '
                  {
                    "a": {
                          "b" : "foo bar",
                          "c" : true
						  "d" :{
                                 "p": "refs/heads/master"
                               }
                         }
                  }
                  '
.\ConsoleApp.exe V --Body $body
```

Here my **C# program** :
```
 [Verb("V")]
    internal class Options
    {
        [Option(Required = true)]
        public string body { get; set; }
    }


    internal class Program
    {
        static void Main(string[] args)
        {
            /*
            Console.WriteLine("----------------");
            Console.WriteLine("COUNT = "+args.Length);
            Console.WriteLine("----------------");
            foreach (var item in args)
                Console.WriteLine(item);
            Console.WriteLine("----------------");*/

            Parser.Default.ParseArguments<Options>(args).MapResult(
                 (Options options) => Run(options),
                 errors => RunErrors(errors));
            if (Debugger.IsAttached)
                Console.ReadLine();
        }

        private static int Run(Options options)
        {
            Console.WriteLine("----------------");
            Console.WriteLine(options.body);
            Console.WriteLine("----------------");
            return 0;
        }

        static internal int RunErrors(IEnumerable<CommandLine.Error> errors)
        {
            Console.WriteLine("Errors count =  "+errors.Count());
            foreach (var item in errors)
            {
                Console.WriteLine(item);
            } 
            return 1;
        }
    }
```

And the **result** is : 
![image](https://user-images.githubusercontent.com/11009997/221888848-7cc7b4d7-f483-4447-be05-bbcecd9cab94.png)

**Is it possible with CommandLineParser** ?
Thanks for your answer...

Regards
Sybaris

**Comment by elgonzo** [comment]
Whether it's possible or not and how it is precisely done depends _entirely_ and _only_ on the shell on which you type/provide the commandline arguments for your program. The library basically just consumes the values passed to the _args_ string array of the _Main_ method, and the library has no control nor influence over how the used shell is passing CLI values to the program and which CLI syntax the shell uses.

So, basically, your question is: Using Powershell, how can i pass a json string with spaces, double quotes _and_ line breaks as a single argument value to some program. That's a question more suitable for a Powershell forum or stackoverflow.com, and not so much for the issue tracker of this library here, unfortunately...

**Comment by sybaris** [comment]
Thanks for your answer.
As suggested, I post the question here : https://stackoverflow.com/questions/75598151/using-powershell-how-can-i-pass-a-json-string-as-a-single-argument-value-to-my-p

1 more question : Does SplitArgs() extension method split like which shell ? SplitArgs() is very usefull to write unit tests

**Comment by ericnewton76** [comment]
Dont use command line arguments.  Use pipes.   Just create a jsonreader from Console.InputStream

```
> type myjson.json | myprog --option1 true --option2 false --destinationUrl "https://localhost:44355/"
```

**Comment by sybaris** [comment]
> Dont use command line arguments. Use pipes. Just create a jsonreader from Console.InputStream
> 
> ```
> > type myjson.json | myprog --option1 true --option2 false --destinationUrl "https://localhost:44355/"
> ```

Thanks for your answer.
Very interesting idea. However, when I do not use pipes (empty argument) "Console.In" prompt data from keyboard. The tool has to run in a non-interactive session...
I change my strategy : My tool will ask for a json filename, and the caller will need to write the file and pass the filename as an argument of the command line.
I feel that I would have less worries this way...

**Comment by elgonzo** [comment]
> However, when I do not use pipes (empty argument)

That's what `Console.IsInputRedirected` is for, so you can check whether stdin has been redirected or not...

**Comment by sybaris** [comment]
I didn't know `Console.IsInputRedirected`. Thanks for the information. Can be usefull...

**Comment by ericnewton76** [comment]
A lot of (*nix) utilities by default just wait for input, so its intended that you'd pipe output from another command into them. `base64` for example, and `sha1sum` commands.  If you run them straight up they just sit there, effectively waiting for the piped output or you to just type something.  It just depends on how you want this to run.  

In this case, its a good option to create a filepath option and read from a file, maybe even set the Console.InputStream to the FileStream and it'll basically ReadToEnd on its own and then you'd check for FileStream.EndOfFile and then terminate.  This is all by memory I havent specifcally written a tool like this in several months but thats the gist of it.
